### PR TITLE
feature/global-login-logout

### PR DESCRIPTION
**Pull request by mtsmachado8**

### CODE REVIEW

File: src/components/Reactions/Reactions.tsx [hunk 1]
```diff
@@ -14,16 +15,48 @@ type FilledComment = Comment & {
...
+const Reactions: React.FC<Props> = (props: Props) => {
+  const { loggedUser } = useContext(AuthContext);
+  const [ alreadyLiked, setAlreadyLiked ] = useState(undefined);
+
+  const likes = props.likes ? props.likes.length : 0;
+
+  useEffect(() => {
+    if(loggedUser) {
+      setAlreadyLiked(
+        props.likes.find(like => like.author.email === loggedUser?.email)
```
Comment (mtsmachado8) [inline src/components/Reactions/Reactions.tsx:39]: you want a boolean here right?
use some
props.likes.some(like => like.author.email === loggedUser?.email)

File: src/components/Reactions/Reactions.tsx [hunk 2]
```diff
@@ -14,16 +15,48 @@ type FilledComment = Comment & {
...
   return(
     <div className={styles.reactions}>
+      <div 
+        className={styles.actions}
+        onClick={e => {
+          e.preventDefault();
+          props.onSubmitNewLike(alreadyLiked ? alreadyLiked.id : null);
+        }}
+      >
+        <BiLike className={alreadyLiked ? styles.unliked : styles.liked} />
```
Comment (mtsmachado8) [inline src/components/Reactions/Reactions.tsx:53]: This loggic doesnt seem right. if i already liked my style will be unliked? what are you trying to say here? try changing it to something like: alreadyLiked ? styles.unlike : styles.like or  styles.alreadyLiked : styles.neverLiked

File: src/components/Reactions/Reactions.tsx [hunk 3]
```diff
@@ -14,16 +15,48 @@ type FilledComment = Comment & {
...
+        <p className={alreadyLiked ? styles.unliked : styles.liked}>{likes}</p>
```
Comment (mtsmachado8) [inline src/components/Reactions/Reactions.tsx:54]: same here

Comment (mtsmachado8) [reply src/components/Reactions/Reactions.tsx:54]: styles.alreadyLiked : styles.neverLiked

File: src/infra/services/alteredLikes.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import api from './api';
+import { trigger } from '../hooks/swr';
+import { toast } from 'react-toastify';
+
+const alteredLabels = async (bubbleId: Number, author: Object, id: Number) => {
```
Comment (mtsmachado8) [inline src/infra/services/alteredLikes.ts:5]: why is it called alteredLikes?

File: src/infra/services/alteredLikes.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  if(author) {
+    if(bubbleId) {
+      if(id) {
+        try {
+          await api.remove(`/likes/${id}`);
+    
+          trigger('/bubbles');
+          trigger(`/bubbles/${bubbleId}`);
+        } catch {
+          toast.error('Alteration error! Please, reload the page and try again', {
```
Comment (mtsmachado8) [inline src/infra/services/alteredLikes.ts:15]: Couldn't unlike bubble! Please, reload the page and try again

File: src/infra/services/alteredLikes.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+          await api.post('/likes', {
+            bubbleId,
+            author,
+          });
+      
+          trigger('/bubbles');
+          trigger(`/bubbles/${bubbleId}`);
+      
+        } catch {
+          toast.error('Alteration error! Please, reload the page and try again', {
```
Comment (mtsmachado8) [inline src/infra/services/alteredLikes.ts:32]: Couldn't like bubble! Please, reload the page and try again

File: src/pages/api/likes/[id].ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { remove } from './_repository';
+
+export default async (req, res) => {
+  if (req.method === 'DELETE') {
+    const { query: { id } } = req;
+
+    const disliked = await remove(
```
Comment (mtsmachado8) [inline src/pages/api/likes/[id].ts:7]: removedLike would be better

File: src/pages/api/likes/_repository.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import prisma from "../../../../prisma/client";
+import { User } from '@prisma/client';
+
+const create = async (bubbleId: number, author: User) => {
+  const liked = await prisma.like.create({
```
Comment (mtsmachado8) [inline src/pages/api/likes/_repository.ts:5]: like would be better. Liked is a verb in the past not a noun

File: src/pages/api/likes/_repository.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+          create: author,
+        },
+      },
+    },
+  });
+  return liked;
+};
+
+const remove = async (id: string) => {
+  const disliked = await prisma.like.delete({
```
Comment (mtsmachado8) [inline src/pages/api/likes/_repository.ts:22]: removedLike

File: src/pages/api/likes/index.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { create } from "./_repository";
+
+export default async (req, res) => {
+  if (req.method === 'POST') {
+    const { bubbleId, author } = req.body;
+
+    const liked = await create(
```
Comment (mtsmachado8) [inline src/pages/api/likes/index.ts:7]: like

File: src/components/Reactions/Reactions.tsx (hunk 1 above)
Comment (MatheusDomingues) [reply src/components/Reactions/Reactions.tsx:39]: On really... not. If I catch just the boolean of this, I can't send the Like id... But I can make another setState, to receive the like details, and this setState, receive just the boolean...

**Pull request merged**